### DISCUSSION TITLE
How are task groups meant to be represented?

### DISCUSSION THREAD

**Discussion by lishaduck** [vercel/turborepo#9323] [Help] [answered]
### Summary

I've been using the following pattern for a while, but the lsp doesn't like. Presumably there's a preferred way of expressing this, so I'm asking:

I've got two tasks  named `typecheck:src` & `typecheck:extra` in both of my sub-`package.json`s. `:src` has the `dom` lib and only targets `src/`, and `:extra` just targets files like `eslint.config.js`. I want to run them (ergonomically) in parallel, so I create a task in my `turbo.json` named `typecheck`. It doesn't actually exist, it just depends on `typecheck:src` and `typecheck:extra`. This works fine, but the lsp complains with a `turbo:no-such-task` error, so presumably there's a better way.
I definitely want to ensure this doesn't break in the future as well (even if not it's recommended, I'd like some way to do this that's not a complete hack, i.e., that turbo won't ban fake tasks)

### Additional information

I can't write this as a task in my root `package.json` b/c I either get a recursive turbo error (when run with Turborepo) or I lose the ability to filter (when running with `pnpm`).

### Example

(None of these are minimal, they all demonstrate workarounds)
- https://github.com/PSDTools/psdtools.github.io (ignore the error)
- https://github.com/jfmengels/node-elm-review (run `typecheck` serially using one command + `&&`)
- https://github.com/AndreaPontrandolfo/sheriff (use separate tasks)

**Comment by anthonyshew** [comment, marked as answer]
What you've written is a good solution. This is a shortcoming of the LSP, and something we should likely work on. If you feel strongly enough that you'd like to create an Issue for the LSP in this case, please do!
